### PR TITLE
proxy refactor serverless

### PR DESCRIPTION
**Pull request by conradludgate**
## Problem

Our serverless backend was a bit jumbled. As a comment indicated, we were handling SQL-over-HTTP in our `websocket.rs` file.

I've extracted out the `sql_over_http` and `websocket` files from the `http` module and put them into a new module called `serverless`.

## Summary of changes

```sh
mkdir proxy/src/serverless
mv proxy/src/http/{conn_pool,sql_over_http,websocket}.rs proxy/src/serverless/
mv proxy/src/http/server.rs proxy/src/http/health_server.rs
mv proxy/src/metrics proxy/src/usage_metrics.rs
```

I have also extracted the hyper server and handler from websocket.rs into `serverless.rs`

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: proxy/src/bin/proxy.rs [hunk 1]
```diff
@@ -49,21 +50,23 @@ async fn main() -> anyhow::Result<()> {
         cancellation_token.clone(),
     ))];
 
-    if let Some(wss_address) = args.get_one::<String>("wss") {
-        let wss_address: SocketAddr = wss_address.parse()?;
-        info!("Starting wss on {wss_address}");
-        let wss_listener = TcpListener::bind(wss_address).await?;
+    // TODO: rename the argument to something like serverless.
+    // It now covers more than just websockets, it also covers SQL over HTTP.
+    if let Some(serverless_address) = args.get_one::<String>("wss") {
```
Comment (skyzh) [inline proxy/src/bin/proxy.rs:55]: we should move to clap4 / structopt someday... 🤣

**Review by skyzh — APPROVED** [review]

**Review by koivunej — APPROVED** [review]
This seems to just shuffle files around and doesn't really have real changes, wonder why github shows the huge diff.

**Comment by conradludgate** [comment]
> wonder why github shows the huge diff.

If you edit the files that change by more than ~10% of lines, it doesn't consider it a move. Since I extracted a large portion of websockets.rs, it get's confused. Looking at the individual commits works better

**Comment by skyzh** [comment]
looks like this PR needs resolving conflicts and auto-merge again 🤪

**Pull request merged**